blog/content/post/2019-05-15-building-bareflank.md

```diff
@@ -0,0 +1,345 @@
+---
+post_title: Building and running Bareflank
+author: krystian.hebel
+layout: post
+published: true
+date: 2019-05-15
+
+tags:
+    - hypervisor
+    - virtualization
+    - bare-metal
+    - VMX
+categories:
+    - Firmware
+
+---
+
+In this second post of a series, we will build and start our first hypervisor.
+It won't do much just yet, but it is good to get known with its build system.
+We will be using [Bareflank](https://github.com/Bareflank/hypervisor).
+
+\################################################################################
+TBD: link first post here
+\################################################################################
+
+## What is Bareflank?
+
+Keep in mind that Bareflank isn't a full hypervisor as such, it is more like
+an SDK for building our own hypervisors with it. In its basic implementation,
+most of the handlers are simple pass-through to the underlying hardware. This
+has some implications, they will be mentioned later.
+
+There are more open-source hypervisors available, such as [SimpleVisor](https://ionescu007.github.io/SimpleVisor/),
+[HyperPlatform](https://github.com/tandasat/HyperPlatform), [ACRN](https://projectacrn.org/)
+or even [Xen](https://xenproject.org/). Bareflank was chosen for this series
+because it is a relatively simple project, providing a friendly API for custom
+handlers, with low hardware and software requirements. We will be running this
+hypervisor on [MinnowBoard Turbot](https://minnowboard.org/minnowboard-turbot/)
+board, but it should work on most UEFI platforms with VMX and [EPT-capable CPU](https://ark.intel.com/content/www/us/en/ark/search/featurefilter.html?productType=873&0_ExtendedPageTables=True).
+
+## Getting the source code
+
+At the moment of writing this blog post, the newest tagged Bareflank version is
+`rc2.0.4`, but soon after it was released there was a big change in the
+structure of this project. Extended APIs (which are required for UEFI build and
+many other things) were merged into the main repository, earlier they were a
+project of its own. This made it difficult to maintain coherent interfaces
+between those two, usually one version of hypervisor was not compatible with
+some versions of extended APIs.
+
+As we really want to use extended APIs, in these series we will use commit
+`Merge pull request #733 from rianquinn/eapis_merge`. It was the latest commit
```

Comment:
@krystian-hebel it would be great to give a link here

Reply:
Done.